src/cocotb/handle.py

```diff
@@ -210,6 +210,15 @@ def __len__(self) -> int:
 KeyType = TypeVar("KeyType")
 
 
+class _GPIDiscovery(enum.IntEnum):
```

Comment:
I think I missed this before, but `_get` is public, this should be as well.

Reply:
Also to get documentation to work you'll probably have to adapt `DocEnum` to an `IntEnum`. I'll link how to do that seeing as your change will probably come in before mine.

Reply:
From #4512 but adapting to `IntEnum` instead of `StrEnum`.

```python
IntEnumT = TypeVar("IntEnumT ", bound=IntEnum)


class DocIntEnum(IntEnum):
    """Like DocEnum but for IntEnum enum types."""

    def __new__(cls: Type[IntEnumT], value: str, doc: Optional[str] = None) -> IntEnumT:
        self = int.__new__(cls, value)
        self._value_ = value
        if doc is not None:
            self.__doc__ = doc
        return self
```